pom.xml

```diff
@@ -24,9 +24,8 @@
   </licenses>
 
   <properties>
-    <jenkins.version>1.625</jenkins.version>
+    <jenkins.version>2.7.1</jenkins.version>
```

Comment:
What is the justification to bump the version so far?

Reply:
This comment and the next one are related, so I'll try to give a common answer to be clearer.

`ssh-slaves` is using the inherited dependency (with test scope) `sshd-core` from the `jenkins-core`. That makes the plugin to be extremely dependent on the `jenkins-core` version. Older versions had a dependency with `sshd-core:0.x`, while `jenkins-core:2.x` have dependency with `sshd-core:1.x`. Many packages and classes have been refactored and moved, so tests would fail depending on the environment we execute them:
- They will succeed in a stand-alone execution or in an environment based on `jenkins-core` versions lower than 2.0.
- They will fail in an environment based on `jenkins-core` version 2.0 or upper.
Trying to avoid those failures, I propose to fix to `ssh-core:1.6.0` (The dependency for newer `jenkins-core` version) and updated the tests. IMO, this will create an actual isolated test environment.
The problem is SSHD module in jenkins-core is using the class `KeyPairProvider` in `sshd-core`, that is one of the refactored classes. Keeping the `jenkins-core:1.625` causes an exception when trying to get the ExtensionList, so it should be upgraded. I have proposed 2.7.1 just because it is a LTS:

`org.jvnet.hudson.reactor.ReactorException: java.lang.Error: java.lang.reflect.InvocationTargetException
        at org.jvnet.hudson.reactor.Reactor.execute(Reactor.java:269)
        at jenkins.InitReactorRunner.run(InitReactorRunner.java:44)
        at jenkins.model.Jenkins.executeReactor(Jenkins.java:916)
        at jenkins.model.Jenkins.<init>(Jenkins.java:815)
        at hudson.model.Hudson.<init>(Hudson.java:85)
        at org.jvnet.hudson.test.JenkinsRule.newHudson(JenkinsRule.java:611)
        at org.jvnet.hudson.test.JenkinsRule.before(JenkinsRule.java:384)
        at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:537)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.Error: java.lang.reflect.InvocationTargetException
        at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:110)
        at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:176)
        at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:282)
        at jenkins.model.Jenkins$7.runTask(Jenkins.java:905)
        at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:210)
        at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        ... 1 more
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.GeneratedMethodAccessor36.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:106)
        ... 8 more
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.rangeCheck(ArrayList.java:653)
        at java.util.ArrayList.get(ArrayList.java:429)
        at hudson.ExtensionList.get(ExtensionList.java:171)
        at hudson.PluginManager$PluginUpdateMonitor.getInstance(PluginManager.java:1258)
        at hudson.maven.PluginImpl.init(PluginImpl.java:58)
        ... 12 more`

I hope to make myself understood.